lib/spack/spack/package_base.py

```diff
@@ -1463,8 +1463,10 @@ def do_fetch(self, mirror_only=False):
             return
 
         checksum = spack.config.get("config:checksum")
+        fetch = self.stage.needs_fetching
         if (
             checksum
+            and fetch
```

Comment:
```suggestion
            and "dev_path" not in self.spec.variants
```

can you simplify to a single line? it's not very different from how git versions are excluded

Reply:
well, move it below the gitversion thing, cause the condition was supposed to be `if <checksum> and <unknown version>`.

Reply:
Done.